source/plugins/go/src/oms.go

```diff
@@ -729,9 +736,22 @@ func flushKubeMonAgentEventRecords() {
 				}
 			}
 			if IsWindows == false && len(msgPackEntries) > 0 { //for linux, mdsd route
-				if IsAADMSIAuthMode == true && strings.HasPrefix(MdsdKubeMonAgentEventsTagName, MdsdOutputStreamIdTagPrefix) == false {
-					Log("Info::mdsd::obtaining output stream id for data type: %s", KubeMonAgentEventDataType)
-					MdsdKubeMonAgentEventsTagName = extension.GetInstance(FLBLogger, ContainerType).GetOutputStreamId(KubeMonAgentEventDataType)
+				if IsAADMSIAuthMode == true {
+					useFromCache := true
+					if !strings.HasPrefix(MdsdKubeMonAgentEventsTagName, MdsdOutputStreamIdTagPrefix) {
```

Comment:
those two conditions are both set the useFromCache = false, can we combine them in a single if and add condition for setting MdsdKubeMonAgentEventsTagRefreshTracker ?